File: src/FactoryMethod/src/models/filme.ts
import { Midia } from './midia';

export class Filme extends Midia {
    private diretor: string;
    private roteirista: string;

    constructor(
        titulo: string,
        descricao: string,
        anoLancamento: number,
        duracao: number,
        genero: string,
        classificacao: string,
        elenco: string[],
        diretor: string,
        roteirista: string
    ) {
        super(titulo, descricao, anoLancamento, duracao, genero, classificacao, elenco);
        this.diretor = diretor;
        this.roteirista = roteirista;
    }

    public printarInformacoes(): void {
        console.log(`Filme: ${this.titulo}`);
        console.log(`Descrição: ${this.descricao}`);
        console.log(`Ano de Lançamento: ${this.anoLancamento}`);
        console.log(`Gênero: ${this.genero}`);
        console.log(`Classificação: ${this.classificacao}`);
        console.log(`Elenco: ${this.elenco.join(', ')}`);
        console.log(`Diretor: ${this.diretor}`);
        console.log(`Roteirista: ${this.roteirista}`);
    }
}

File: src/FactoryMethod/src/Filme.java
import java.util.ArrayList;

public class Filme extends Midia {
    private String diretor;
    private String roteirista;

    public Filme(String titulo, String descricao, int anoLancamento, int duracao, String genero, String classificacao, ArrayList<String> elenco) {
        super(titulo, descricao, anoLancamento, duracao, genero, classificacao, elenco);
    }

    public void printarInformacoes() {
        System.out.println("Filme: " + titulo);
        System.out.println("Descrição: " + descricao);
        System.out.println("Ano de Lançamento: " + anoLancamento);
        System.out.println("Gênero: " + genero);
        System.out.println("Classificação: " + classificacao);
        System.out.println("Elenco: " + (elenco != null ? String.join(", ", elenco) : "N/A"));
    }
}
